deploy/deploy_test.go

```diff
@@ -416,7 +523,7 @@ func TestMetalLBSpec(t *testing.T) {
 		},
 	}
 	mockCtrl := gomock.NewController(t)
-	defer mockCtrl.Finish()
+	mockCtrl.Finish()
```

Comment:
reason for changing this?

Reply:
Nope, that was from when I was trying to figure out why there were mock failures and I failed to change it back.